cmake/compiler.cmake

```diff
@@ -0,0 +1,45 @@
+include(CheckCCompilerFlag)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_FLAGS_DEBUG "-O0 -g -ggdb -DDEBUG")
+set(CMAKE_C_FLAGS_RELEASE "-O2 -flto")
+set(CMAKE_INTERPROCEDURAL_OPTMIZATION TRUE)
+
+function(add_compile_option_if option variable)
```

Comment:
Ahhh, so this basically checks if the option is available before enabling it?  Sounds good